resources/server/twitter/twitter.service.ts

```diff
@@ -25,60 +30,56 @@ class _TwitterService {
       // We should be able to comment this out as profile **should** be guranteed,
       // as we create a default profile in that process.
 
-      // if (!profile) {
-      //   const defaultProfileNames = await getDefaultProfileNames(identifier);
-      //   emitNet(TwitterEvents.GET_OR_CREATE_PROFILE_NULL, _source, defaultProfileNames);
-      // } else {
-      emitNet(TwitterEvents.GET_OR_CREATE_PROFILE_SUCCESS, src, profile);
-      // }
+      if (!profile) {
+        const defaultProfileNames = await getDefaultProfileNames(reqObj.source);
+
+        console.log('We have no profile');
+        console.log(defaultProfileNames);
+
+        emitNet(TwitterEvents.GET_OR_CREATE_PROFILE_NULL, reqObj.source, defaultProfileNames);
+      } else {
+        resp({ status: 'ok', data: profile });
+      }
     } catch (e) {
-      emitNet(TwitterEvents.GET_OR_CREATE_PROFILE_FAILURE, src);
+      emitNet(TwitterEvents.GET_OR_CREATE_PROFILE_FAILURE, reqObj.source);
       twitterLogger.error(`Failed to get or create profile, ${e.message}`, {
-        source: src,
+        source: reqObj.source,
       });
     }
   }
 
-  async handleCreateProfile(src: number, profile: Profile) {
+  async handleCreateProfile(reqObj: PromiseRequest<Profile>, resp: PromiseEventResp<void>) {
     try {
-      const identifier = PlayerService.getIdentifier(src);
-      await this.twitterDB.createProfile(identifier, profile.profile_name);
+      const identifier = PlayerService.getIdentifier(reqObj.source);
+      await this.twitterDB.createProfile(identifier, reqObj.data.profile_name);
 
-      emitNet(TwitterEvents.CREATE_PROFILE_RESULT, src, {
-        message: 'TWITTER_CREATE_PROFILE_SUCCESS',
-        type: 'success',
-      });
+      // should we get the profile here?
+      resp({ status: 'ok' });
```

Comment:
> // should we get the profile here?
 
I agree with this comment wholeheartedly. Its pretty typical for an API to return final data after a mutation operation.